lib/src/components/control/scrollview.dart

```diff
@@ -1,23 +1,38 @@
 import 'dart:math';
-
 import 'package:flutter/gestures.dart';
 import 'package:flutter/scheduler.dart';
 import 'package:shadcn_flutter/shadcn_flutter.dart';
 
+/// ScrollInterCeptionBehavior is used to decide scroll behavior by using hover or drag interactions.
+class ScrollInterCeptionBehavior {
+  /// Whether scroll interception is enabled.
+  final bool enabled;
+
+  /// disables chrome like mmb toggle to activate scrolling on mouse movement.
+  final bool holdDrag;
+
+  /// By default scroll interception activated by dragging mmb interaction.
+  const ScrollInterCeptionBehavior(
+      {required this.enabled, this.holdDrag = true});
```

Comment:
I think `this.enabled` should not be required and has default set to `false`
```suggestion
      {this.enabled = false, this.holdDrag = true});
```